packages/ng-primitives/dialog/src/dialog-overlay/dialog-overlay.ts

```diff
@@ -21,6 +21,23 @@ export class NgpDialogOverlay {
     transform: booleanAttribute,
   });
 
+  /**
+   * Whether the dialog should close on escape key press.
+   * @default `false`
+   */
+  readonly disableEscapeKey = input<boolean, BooleanInput>(false, {
+    alias: 'ngpDialogOverlayDisableEscapeKey',
+    transform: booleanAttribute,
+  });
+
+  disableEscapeKeyEffect = effect(() => {
+    if (this.disableEscapeKey()) {
+      this.dialogRef.disableEscapeKey = true;
+    } else {
+      this.dialogRef.disableEscapeKey = false;
+    }
+  });
```

Comment:
I had some time to look into this further - and I think it would be better if this was moved to the dialog-trigger component - I told you to put it here, so that was bad advice on my part. It makes sense for click as the click is actually occurring on the backdrop, but the escape key listening is not an overlay thing. If we move it to the trigger we can simply pass it as config to the `open` method, which will mean we don't need any effects or things like that.

I hope that makes sense.